BUILD.bazel

```diff
@@ -0,0 +1,2492 @@
+load("@rules_proto//proto:defs.bzl", "proto_library")
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_proto_library", "cc_test")
+load("//third_party:substitution.bzl", "template_rule")
+load("//tools/rules:cu.bzl", "cu_library")
+load("//tools/config:defs.bzl", "if_cuda")
+load("//:aten.bzl", "intern_build_aten_ops")
+
+COMMON_COPTS = [
+    "-DHAVE_MALLOC_USABLE_SIZE=1",
+    "-DHAVE_MMAP=1",
+    "-DHAVE_SHM_OPEN=1",
+    "-DHAVE_SHM_UNLINK=1",
+    "-D_FILE_OFFSET_BITS=64",
+    "-DHAVE_GCC_GET_CPUID",
+    "-DUSE_GCC_GET_CPUID",
+    "-DTH_HAVE_THREAD",
+    "-DUSE_FBGEMM",
+    "-DUSE_DISTRIBUTED",
+    "-DAT_PARALLEL_NATIVE=1",
+    "-DATEN_THREADING=NATIVE",
+    "-DNO_CUDNN_DESTROY_HANDLE",
+] + if_cuda([
+    "-DUSE_CUDA",
+    "-DUSE_CUDNN",
+])
+
+# c10
+template_rule(
+    name = "cmake_macros_h",
+    src = "c10/macros/cmake_macros.h.in",
+    out = "c10/macros/cmake_macros.h",
+    substitutions = {
+        "cmakedefine": "define",
+        "#define FEATURE_TORCH_MOBILE": "/* #undef FEATURE_TORCH_MOBILE */",
+        "#define USE_STATIC_DISPATCH": "/* #undef USE_STATIC_DISPATCH */",
+        "#define C10_USE_NUMA": "/* #undef C10_USE_NUMA */",
+    },
+)
+
+template_rule(
+    name = "cuda_cmake_macros_h",
+    src = "c10/cuda/impl/cuda_cmake_macros.h.in",
+    out = "c10/cuda/impl/cuda_cmake_macros.h",
+    substitutions = {
+        "cmakedefine": "define",
+    },
+)
+
+cc_library(
+    name = "c10_headers",
+    hdrs = glob([
+        "c10/core/*.h",
+        "c10/core/impl/*.h",
+        "c10/cuda/*.h",
+        "c10/cuda/impl/*.h",
+        "c10/macros/*.h",
+        "c10/util/*.h",
+    ]) + [
+        "c10/macros/cmake_macros.h",
+        "c10/cuda/impl/cuda_cmake_macros.h",
+    ],
+    deps = [
+        "@com_github_gflags_gflags//:gflags",
```

Comment:
I wonder how you can get by with this. `c10_headers` in linked in one of the .sos that have linkstatic, and my build complains about duplicate flags definitions.